front/app/containers/Authentication/steps/Verification/VerificationSteps.tsx

```diff
@@ -27,67 +29,84 @@ const Container = styled.div`
 export interface Props {
   onCompleted?: () => void;
   onError?: () => void;
+  authenticationData: AuthenticationData;
 }
 
-const VerificationSteps = memo<Props>(({ onCompleted, onError }) => {
-  const [activeStep, setActiveStep] =
-    useState<TVerificationStep>('method-selection');
-  const [method, setMethod] = useState<TVerificationMethod | null>(null);
-
-  const { data: authUser } = useAuthUser();
-  const { data: verificationMethods } = useVerificationMethods();
-
-  useEffect(() => {
-    if (activeStep === 'success' && onCompleted) {
-      onCompleted();
-    }
-
-    if (activeStep === 'error' && onError) {
-      onError();
-    }
-  }, [onCompleted, onError, activeStep]);
-
-  const onMethodSelected = (selectedMethod: TVerificationMethod) => {
-    setMethod(selectedMethod);
-    setActiveStep('method-step');
-  };
-
-  const goToSuccessStep = useCallback(() => {
-    if (!isNilOrError(authUser)) {
-      invalidateQueryCache();
-      setActiveStep('success');
+const VerificationSteps = memo<Props>(
+  ({ onCompleted, onError, authenticationData }) => {
+    const [activeStep, setActiveStep] =
+      useState<TVerificationStep>('method-selection');
+    const [method, setMethod] = useState<TVerificationMethod | null>(null);
+
+    const { data: authUser } = useAuthUser();
+    const { data: verificationMethods } = useVerificationMethods();
+
+    useEffect(() => {
+      if (activeStep === 'success' && onCompleted) {
+        onCompleted();
+      }
+
+      if (activeStep === 'error' && onError) {
+        onError();
+      }
+    }, [onCompleted, onError, activeStep]);
+
+    const onMethodSelected = (selectedMethod: TVerificationMethod) => {
+      // Save the successAction and the current context in local
+      // storage for when user returns from verification.
+      if (authenticationData?.successAction) {
+        localStorage.setItem(
+          'auth_success_action',
+          JSON.stringify(authenticationData.successAction)
+        );
+      }
+      if (authenticationData?.context) {
+        localStorage.setItem(
+          'auth_context',
+          JSON.stringify(authenticationData.context)
+        );
```

Comment:
Even if a verification method doesn't leave our platform, I don't believe there is any harm saving this to local storage anyways right? :thinking: Since when the `useSteps` picks up the flow again and the `verification_success` param is true, it will get removed from local storage (and I'm assuming the `verification_success` would be set for all methods of verification even if it doesn't leave the platform?)

Reply:
> and I'm assuming the verification_success would be set for all methods of verification even if it doesn't leave the platform?

Nope, it only gets set when you get redirected back to the platform. If you never leave the platform it never gets set.

Purely functionally it will won't make a difference, but it's just cleaner to not write things into people's local storage if it's not really needed and there is an easy way to avoid it (which there is in this case)